lib/Varien/File/Uploader.php

```diff
@@ -504,9 +504,9 @@ private function _setUploadFileId($fileId)
             $this->_uploadType = self::MULTIPLE_STYLE;
             $this->_file = $fileId;
         } else {
-            preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);
+            $result = preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);
 
-            if (count($file) > 0 && (count($file[0]) > 0) && (count($file[1]) > 0)) {
+            if ($result && count($file) === 3) {
```

Comment:
Why ==3 ? Previous code checked only whether first two items are

Reply:
index 0 is the whole string matched, 1 is the first capture and 2 is the second capture, so three elements. The full matched string is dropped on the next line by array_shift. I kinda think the regex should be changed to be `.+?` rather than `.*?` so that all captures must be non-empty.

Reply:
```suggestion
            if (preg_match("/^(.+?)\[(.+?)\]$/", $fileId, $file)) {
```